pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -100,15 +102,35 @@ func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, uid stri
 		return nil
 	}
 
+	cache := PodCacheInterface{Iface: iface}
+
+	indexIPv4 := 0
+	indexIPv6 := 1
+	if !Handler.IsIpv4Primary() {
+		indexIPv4 = 1
+		indexIPv6 = 0
+	}
+
 	for _, addr := range addrList {
 		if addr.IP.IsGlobalUnicast() {
-			err = writeToCachedFile(PodCacheInterface{Iface: iface, PodIP: addr.IP.String()},
-				util.VMIInterfacepath, uid, iface.Name)
-			if err != nil {
-				log.Log.Reason(err).Errorf("failed to write pod Interface to cache, %s", err.Error())
-				return err
+			addrString := addr.IP.String()
+			index := indexIPv4
+			if netutils.IsIPv6String(addrString) {
+				index = indexIPv6
 			}
-			return nil
+
+			cache.PodIPs[index] = addrString
+			if index == 0 {
+				cache.PodIP = addrString
+			}
+		}
+	}
+
+	if cache.PodIP != "" {
```

Comment:
Why would `cache.PodIP`. I you believe it can happen, please add else returning an error.